File: platforms/hacker-earth/count-valid-pairs-2-da02a8f2/solutions/solution.go
package main

import "fmt"

func numsum(n int) int {
	s := 0
	for n > 0 {
		s += n % 10
		n = n / 10
	}
	return s
}

func solve(n int, nums []int) int64 {
	count := make(map[int]int)
	for _, n := range nums {
		s := numsum(n)
		count[s]++
	}

	pairs := 0
	for _, v := range count {
		pairs += v * (v - 1) / 2
		// fmt.Println(k, v, pairs)
	}
	return int64(pairs)
}

func main() {

	var n int
	fmt.Scanln(&n)
	nums := make([]int, n)
	for i_nums := 0; i_nums < n; i_nums++ {
		fmt.Scan(&nums[i_nums])
	}

	var out_ int64 = solve(n, nums)
	fmt.Print(out_)
}

File: platforms/hacker-earth/count-valid-pairs-2-da02a8f2/solutions/solution.py
from collections import Counter


def numsum(n):
    s = 0
    while n > 0:
        s += n % 10
        n //= 10
    return s


def solve(n, nums):
    count = Counter()

    for e in nums:
        count[numsum(e)] += 1

    res = 0
    for v in count.values():
        res += v * (v - 1) // 2
    return res


n = int(input())
nums = list(map(int, input().split()))

out_ = solve(n, nums)
print(out_)
